libraries/rush-lib/src/logic/operations/DependencyAnalysisPlugin.ts

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as os from 'os';
+import { ITerminal } from '@rushstack/node-core-library';
+import {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { IExecutionResult } from './IOperationExecutionResult';
+
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import { IPackageJsonDependencyTable, JsonFile } from '@rushstack/node-core-library';
+import { IBuildTimeRecord, _setBuildTimes } from './BuildTimePlugin';
+import { ITelemetryData } from '../Telemetry';
+
+const PLUGIN_NAME: 'DependencyAnalysisPlugin' = 'DependencyAnalysisPlugin';
+const MAX_CORES: number = 128;
+
+interface IBuildTimeSavedRecord {
+  project: string;
+  secondsSaved: number;
+}
+
+export interface IDependencyGraphEntry {
+  dependencies: string[];
+  dependents: string[];
```

Comment:
Because the edit distance between `dependency` and `dependent` is so small, I recommend using `consumer` as the opposite of `dependency`.

Reply:
Good call. Thanks